In [1]:
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
%config InlineBackend.figure_format = 'svg'
from scipy import stats

##### 1.Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.

In [2]:
y1 = np.array([173, 175, 180, 178, 177, 185, 183, 182])
y2 = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
y3 = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

Гипотеза H0:
вид спорта не оказывает статистически значимого влияния на рост спортсменов$$
$$Гипотеза H1:
вид спорта оказывает статистически значимое влияние на рост спортсменов

In [3]:
y = np.concatenate([y1, [0], y2, [0], y3])
n = len(y)
n1 = y1.size
n2 = y2.size
n3 = y3.size
k = 3
# средне групповые значения
y1_mean = y1.mean()
y2_mean = y2.mean()
y3_mean = y3.mean()

# общее среднее 
y_all = np.concatenate([y1, y2, y3])
y_mean = y_all.sum() / y_all.size

print (f'y1_mean = {round(y1_mean, 3)}')
print (f'y2_mean = {round(y2_mean, 3)}')
print (f'y3_mean = {round(y3_mean, 3)}\n')

print (f'y_mean  = {round(y_mean, 3)}\n')

#----------------------------------------------------------------------------------------------------
# сумма квадратов отклонений наблюдений от общего среднего
s2 = np.sum( (y_all - y_mean)**2 )

# сумма квадратов отклонений средних групповых значений от общего среднего
s2_f = ((y1_mean - y_mean)**2) * n1 + \
        ((y2_mean - y_mean)**2) * n2 + \
        ((y3_mean - y_mean)**2) * n3

# остаточная сумма квадратов отклонений
s2_res = np.sum((y1 - y1_mean)**2) + \
                np.sum((y2 - y2_mean)**2) + \
                np.sum((y3 - y3_mean)**2)

print (f's2          = {round(s2, 3)}')
print (f's2_f        = {round(s2_f, 3)}')
print (f's2_resl = {round(s2_res, 3)}\n')

#----------------------------------------------------------------------------------------------------
# соблюдение равенства: s2 = s2_f + s2_res
print(f's2 = s2_f + s2_res: {round(s2, 3)} = {round(s2_f, 3)} + {round(s2_res, 3)} = \
{round(s2_f, 3) + round(s2_res, 3)}\n')

#----------------------------------------------------------------------------------------------------
# общая дисперсия
sigma2_general = s2 / (n - 1)

# факторная дисперсия
sigma2_f = s2_f / (k - 1)

# остаточная дисперсия
sigma2_res = s2_res / (n - k)

print (f'sigma2_general  = {round(sigma2_general, 3)}')
print (f'sigma2_f        = {round(sigma2_f, 3)}')
print (f'sigma2_res = {round(sigma2_res, 3)}\n')

#----------------------------------------------------------------------------------------------------
# критерий Фишера расчетный
Fp = sigma2_f / sigma2_res
print (f'Fp = {round(Fp, 3)}')

y1_mean = 179.125
y2_mean = 178.667
y3_mean = 172.727

y_mean  = 176.464

s2          = 830.964
s2_f        = 253.907
s2_resl = 577.057

s2 = s2_f + s2_res: 830.964 = 253.907 + 577.057 = 830.964

sigma2_general  = 28.654
sigma2_f        = 126.954
sigma2_res = 21.372

Fp = 5.94


Найдем значение F_крит в табюлице критических точек распредления Фишера для заданного уровня значимолсти alpha = 0,05 и двух степеней свободы$$
df_межд = k-1 = 3-1=2$$
$$df_внут = n-k = 28-3 =25$$
F_крит = 3.38$$
5.94 > 3.38 $$
Верна гипотеза H1:Вид спорта ОКАЗЫВАЕТ статистически значимое вляние на рост спортсмена на уровне значимости 5%